src/server/serve.rs

```diff
@@ -176,7 +178,7 @@ where
                 continue;
             }
         };
-
+        tokio::spawn(monitor_upstream_resolvers(nameservers.clone()));
```

Comment:
This is the wrong place to do this, essentially you create a task leak because every time aardvark-dns gets reloaded (i.e. each container start/stop) you create a new task but the older one never terminates.
this would need to happen in serve() before the loop to only start it once.